app/src/main/java/com/kickstarter/viewmodels/projectpage/ProjectPageViewModel.kt

```diff
@@ -347,86 +356,93 @@ interface ProjectPageViewModel {
         @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
         val onThirdPartyEventSent = BehaviorSubject.create<Boolean?>()
 
+        val disposables = CompositeDisposable()
         init {
 
             val progressBarIsGone = PublishSubject.create<Boolean>()
 
             val mappedProjectNotification = Observable.merge(
-                intent(),
-                intent()
-                    .compose(takeWhen<Intent, Void>(this.reloadProjectContainerClicked))
-            )
-                .switchMap {
-                    ProjectIntentMapper.project(it, this.apolloClient)
-                        .doOnSubscribe {
-                            progressBarIsGone.onNext(false)
-                        }
-                        .doAfterTerminate {
-                            progressBarIsGone.onNext(true)
-                        }
-                        .withLatestFrom(currentConfig.observable(), currentUser.observable()) { project, config, user ->
-                            return@withLatestFrom project.updateProjectWith(config, user)
-                        }
-                        .materialize()
-                }
+                intent,
+                intent
+                    .compose(
+                        takeWhenV2(
+                            this.reloadProjectContainerClicked
+                        )
+                    )
+            ).switchMap {
+                ProjectIntentMapper.project(it, this.apolloClient)
+                    .doOnSubscribe {
+                        progressBarIsGone.onNext(false)
+                    }
+                    .doAfterTerminate {
+                        progressBarIsGone.onNext(true)
+                    }
+                    .withLatestFrom(currentConfig.observable(), currentUser.observable()) { project, config, user ->
+                        return@withLatestFrom project.updateProjectWith(config, user.getValue())
+                    }
+                    .materialize()
+            }
                 .share()
 
-            activityResult()
+            activityResult
                 .filter { it.isOk }
                 .filter { it.isRequestCode(ActivityRequestCodes.SHOW_REWARDS) }
-                .compose(bindToLifecycle())
                 .subscribe { this.expandPledgeSheet.onNext(Pair(true, true)) }
+                .addToDisposable(disposables)
 
-            intent()
```

Comment:
just an fyi, your approach works perfectly, but in order reduce amount of code changes, take a look for example at [MessagesActivity](https://github.com/kickstarter/android-oss/blob/27b590b8d6f7e5f6ee4192eaa6970c4192b0e7c6/app/src/main/java/com/kickstarter/viewmodels/MessagesViewModel.kt#L586)

The intent can be passed to the viewmodelProvider factory, and then have a private method 
`private fun intent() = intent?.let { Observable.just(it) } ?: Observable.empty()`

in that way you don't need to create the new input `fun configureWith(intent: Intent)`